src/coreComponents/constitutive/fluid/multifluid/blackOil/BlackOilFluidBase.cpp

```diff
@@ -232,8 +253,10 @@ void BlackOilFluidBase::createAllKernelWrappers()
       validateTable( viscosityTable, true );
 
       // create the table wrapper for the oil and (if present) the gas phases
-      m_formationVolFactorTables.emplace_back( fvfTable.createKernelWrapper() );
-      m_viscosityTables.emplace_back( viscosityTable.createKernelWrapper() );
+      m_formationVolFactorTables.emplace_back( &fvfTable );
+      m_viscosityTables.emplace_back( &viscosityTable );
```

Comment:
This can be done at the beginning of the simulation and not necessarily when we create the kernel wrappers

Reply:
Does this last commit match what you had in mind?